### ISSUE TITLE
EnableEndpointRouting can't generate url

### ISSUE THREAD

**Issue by SeriaWei** [aspnet/AspNetCore#7120]
Route template:
```
app.UseMvc(routes =>
{
    routes.MapRoute(
            name: "c-default",
            template: "{culture}/{controller}/{action}", 
            defaults: new { controller = "Home", action = "Index" }, 
            constraints: new { culture = new CultureConstraint() });               
});
```
CultureConstraint:
```
public class CultureConstraint : IRouteConstraint
{
    public bool Match(HttpContext httpContext, IRouter route, string routeKey, RouteValueDictionary values, RouteDirection routeDirection)
    {
        return true;
    }
}
```
Use Url.Action to generate two links in HomeController Index view:
```
<a class="nav-link text-dark" href="@Url.Action("Index")">Home</a>
<a class="nav-link text-dark" href="@Url.Action("Privacy")">Privacy</a>
```
PS: Privacy action is also in HomeController

Run the application then visit `/us`, `Privacy` url is not generated:
![image](https://user-images.githubusercontent.com/6006218/51968747-abb1a900-24ad-11e9-9d26-30c5a3464d38.png)

But, if you pass into `culure` to `Url.Action` method, the url can be generated correctly:
```
Url.Action("Privacy",new { culture="us" })
```

And if disable endpoint routing, the url can be generated correctly without pass into culture:
```
services.AddMvc(options =>
{
    options.EnableEndpointRouting = false;
}).AddControllersAsServices().SetCompatibilityVersion(CompatibilityVersion.Latest);
```
![image](https://user-images.githubusercontent.com/6006218/51969075-75c0f480-24ae-11e9-82b7-756347c2cb7a.png)

**Comment by SeriaWei** [comment]
Is this a bug or the endpoint routing is just like that?

**Comment by Timynore** [comment]
I'm having this problem too and disable endpoint routing temporary to make the url correct.

**Comment by mkArtakMSFT** [comment]
@rynowak  can you please answer this? Thanks!

**Comment by SeriaWei** [comment]
Can anyone help me answer this? Thanks so much!

**Comment by rynowak** [comment]
I tested this out, and it's a bug in 2.2 that has already been fixed for 3.0. In 2.2 we do some post-processing of routes which causes artifacts like this. The feature has been redesigned for 3.0 and is now much more compatibile.

For the time being on 2.2 I would suggest turning off Endpoint Routing and using the old routing system. 

Updated with equivalent 3.0 code sample.

```C#
            app.UseEndpoints(endpoints =>
            {
                endpoints.MapControllerRoute(
                        name: "c-default",
                        pattern: "{culture}/{controller}/{action}",
                        defaults: new { controller = "Home", action = "Index" },
                        constraints: new { culture = new CultureConstraint() });
            });
```
